File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--binding-data-to-controls-on-a-word-actions-pane_5.vb
    Private Sub Insert_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Insert.Click

        Dim tbl As System.Data.DataTable = NorthwindDataSet.Products
        Dim rows() As System.Data.DataRow

        ' Check if a product is selected.
        If Not Me.ProductNameListBox.SelectedIndex < 0 Then

            Dim product As String = Me.ProductNameListBox.SelectedItem.Row.Item("ProductName")
            Dim company As String = Me.CompanyNameComboBox.Text

            ' Return the data row from the selected Product in the list box.
            rows = tbl.Select("[ProductName] = '" & product.Replace("'", "''") & "'")

            AddData(rows(0), company)

        Else
            MessageBox.Show("Please select a product.", "Actions Pane", MessageBoxButtons.OK)
        End If
    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--binding-data-to-controls-on-a-word-actions-pane_5.cs
        private void Insert_Click(object sender, System.EventArgs e)
        {
            System.Data.DataTable tbl = northwindDataSet.Products;
            System.Data.DataRow[] rows;

            // Check if a product is selected.
            if (this.productNameListBox.SelectedIndex >= 0)
            {
                System.Data.DataRowView productRow = (System.Data.DataRowView)this.productNameListBox.SelectedItem;

                string product = productRow.Row["ProductName"].ToString();
                string company = this.companyNameComboBox.Text;

                // Return the data row from the selected product.
                rows = tbl.Select("[ProductName] = '" + product.Replace("'", "''") + "'");

                this.AddData(rows[0], company);
            }
            else
            {
                MessageBox.Show("Please select a product.", "Actions Pane", MessageBoxButtons.OK);
            }
        }